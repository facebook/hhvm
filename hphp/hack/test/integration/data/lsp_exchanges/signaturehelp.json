[
  {
    "jsonrpc": "2.0",
    "id": 1,
    "method": "initialize",
    "params": {
      "initializationOptions": {},
      "processId": null,
      "rootPath": "${root_path}",
      "capabilities": {}
    }
  },
  {
    "jsonrpc": "2.0",
    "method": "textDocument/didOpen",
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}",
        "languageId": "hack",
        "version": 1,
        "text": "${php_file}"
      }
    }
  },
  {
    "comment": "signature help for 0-argument method invocation (left of opening paren)",
    "jsonrpc": "2.0",
    "method": "textDocument/signatureHelp",
    "id": 10,
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 4,
        "character": 23
      }
    }
  },
  {
    "comment": "signature help for 0-argument method invocation",
    "jsonrpc": "2.0",
    "method": "textDocument/signatureHelp",
    "id": 20,
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 4,
        "character": 24
      }
    }
  },
  {
    "comment": "signature help for 0-argument method invocation (right of closing paren)",
    "jsonrpc": "2.0",
    "method": "textDocument/signatureHelp",
    "id": 30,
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 4,
        "character": 25
      }
    }
  },
  {
    "comment": "signature help for non-0-argument function invocation (left of opening paren)",
    "jsonrpc": "2.0",
    "method": "textDocument/signatureHelp",
    "id": 40,
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 5,
        "character": 15
      }
    }
  },
  {
    "comment": "signature help for non-0-argument function invocation (right of opening paren)",
    "jsonrpc": "2.0",
    "method": "textDocument/signatureHelp",
    "id": 50,
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 5,
        "character": 16
      }
    }
  },
  {
    "comment": "signature help for non-0-argument function invocation (left of first comma)",
    "jsonrpc": "2.0",
    "method": "textDocument/signatureHelp",
    "id": 60,
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 5,
        "character": 26
      }
    }
  },
  {
    "comment": "signature help for non-0-argument function invocation (right of first comma)",
    "jsonrpc": "2.0",
    "method": "textDocument/signatureHelp",
    "id": 70,
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 5,
        "character": 27
      }
    }
  },
  {
    "comment": "signature help for non-0-argument function invocation (left of closing paren)",
    "jsonrpc": "2.0",
    "method": "textDocument/signatureHelp",
    "id": 80,
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 5,
        "character": 30
      }
    }
  },
  {
    "comment": "signature help for non-0-argument function invocation (right of closing paren)",
    "jsonrpc": "2.0",
    "method": "textDocument/signatureHelp",
    "id": 90,
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 5,
        "character": 31
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 999,
    "method": "shutdown",
    "params": {}
  }
]
