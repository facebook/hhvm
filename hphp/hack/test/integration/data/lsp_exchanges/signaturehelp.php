<?hh // strict

/** Function doc block */
function function_call(MyClass $myObject, int $x): void {
  $myObject->methodCall();
  function_call($myObject , $x);
}

class MyClass {
  /** Method doc block */
  public function methodCall(): void {}
}
