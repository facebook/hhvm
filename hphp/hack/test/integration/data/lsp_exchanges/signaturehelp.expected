[
  {
    "jsonrpc": "2.0",
    "id": 1,
    "result": {
      "capabilities": {
        "textDocumentSync": {
          "openClose": true,
          "change": 2,
          "willSave": false,
          "willSaveWaitUntil": false,
          "save": {
            "includeText": false
          }
        },
        "hoverProvider": true,
        "completionProvider": {
          "resolveProvider": true,
          "triggerCharacters": [
            "$",
            ">",
            "\\",
            ":",
            "<",
            "[",
            "'",
            "\""
          ]
        },
        "signatureHelpProvider": {
          "triggerCharacters": [
            "(",
            ","
          ]
        },
        "definitionProvider": true,
        "referencesProvider": true,
        "documentHighlightProvider": true,
        "documentSymbolProvider": true,
        "workspaceSymbolProvider": true,
        "codeActionProvider": false,
        "documentFormattingProvider": true,
        "documentRangeFormattingProvider": true,
        "documentOnTypeFormattingProvider": {
          "firstTriggerCharacter": ";",
          "moreTriggerCharacter": [
            "}"
          ]
        },
        "renameProvider": true,
        "typeCoverageProvider": true,
        "rageProvider": true
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 10,
    "result": null
  },
  {
    "jsonrpc": "2.0",
    "id": 20,
    "result": {
      "signatures": [
        {
          "label": "public function methodCall(): void",
          "documentation": "Method doc block",
          "parameters": []
        }
      ],
      "activeSignature": 0,
      "activeParameter": 0
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 30,
    "result": null
  },
  {
    "jsonrpc": "2.0",
    "id": 40,
    "result": null
  },
  {
    "jsonrpc": "2.0",
    "id": 50,
    "result": {
      "signatures": [
        {
          "label": "function function_call(MyClass $myObject, int $x): void",
          "documentation": "Function doc block",
          "parameters": [
            {
              "label": "$myObject"
            },
            {
              "label": "$x"
            }
          ]
        }
      ],
      "activeSignature": 0,
      "activeParameter": 0
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 60,
    "result": {
      "signatures": [
        {
          "label": "function function_call(MyClass $myObject, int $x): void",
          "documentation": "Function doc block",
          "parameters": [
            {
              "label": "$myObject"
            },
            {
              "label": "$x"
            }
          ]
        }
      ],
      "activeSignature": 0,
      "activeParameter": 0
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 70,
    "result": {
      "signatures": [
        {
          "label": "function function_call(MyClass $myObject, int $x): void",
          "documentation": "Function doc block",
          "parameters": [
            {
              "label": "$myObject"
            },
            {
              "label": "$x"
            }
          ]
        }
      ],
      "activeSignature": 0,
      "activeParameter": 1
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 80,
    "result": {
      "signatures": [
        {
          "label": "function function_call(MyClass $myObject, int $x): void",
          "documentation": "Function doc block",
          "parameters": [
            {
              "label": "$myObject"
            },
            {
              "label": "$x"
            }
          ]
        }
      ],
      "activeSignature": 0,
      "activeParameter": 1
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 90,
    "result": null
  },
  {
    "jsonrpc": "2.0",
    "id": 999,
    "result": null
  }
]